python/CMakeLists.txt

```diff
@@ -6,9 +6,9 @@ include_directories (${NUMPY_INCLUDE_DIRS})
 include_directories (${Boost_INCLUDE_DIRS})
 
 python_add_module (lcmaes lcmaes.cc)
-target_link_libraries(lcmaes cmaes ${Boost_LIBRARIES})
+target_link_libraries(lcmaes cmaes ${Boost_LIBRARIES} ${LIBCMAES_DEPS})
 if (LINK_PYTHON)
-  target_link_libraries(lcmaes ${PYTHON_LIBRARIES})
+  target_link_libraries(lcmaes ${PYTHON_LIBRARIES} ${LIBCMAES_DEPS})
 endif ()
 
```

Comment:
It should not be necessary to link to tbb here.

Reply:
> It should not be necessary to link to tbb here.

It is. Because `lcmaes.cc` includes `cmaes.h` and we have the definition `USE_TBB` enabled. Nevertheless, I agree with you that using TBB inside python (or even OpenMP) is not a good idea. We could write it in the readme that if you want to use the python bindings, disable TBB and OpenMP.

Reply:
> It should not be necessary to link to tbb here.

You're right.. I will remove it..

Reply:
Then we should make the use of tbb private: dont expose it in public headers but move the definitions that use tbb in cpp files if possible.
Anyways you're linking twice here in the if (LINK_PYTHON).